File: javascript/day-2.js
const fs = require('fs')
const data = fs.readFileSync('inputs/day-2.txt').toString().split('\n')
for (i = 0; i < data.length; i++) {
    data[i] = data[i].split(/ +/g)
}

let hPos = 0
let depth = 0
let aim = 0

// Part one
data.forEach(step => {
    if (step[0] == 'forward') {
        hPos += Number(step[1])
    } else if (step[0] == 'down') {
        depth += Number(step[1])
    } else {
        depth -= Number(step[1])
    }
})
console.log(hPos * depth)

// Part two
hPos = 0
depth = 0
data.forEach(step => {
    if (step[0] == 'forward') {
        hPos += Number(step[1])
        depth += aim * Number(step[1])
    } else if (step[0] == 'down') {
        aim += Number(step[1])
    } else {
        aim -= Number(step[1])
    }
})
console.log(hPos * depth)

File: 2021/day-2.py
with open('inputs/day-2.txt') as file:
    data = file.read().split('\n')
    data = [x.split() for x in data]

# Part one
h_pos = 0
depth = 0
for step in data:
    if step[0] == 'forward':
        h_pos += int(step[1])
    elif step[0] == 'down':
        depth += int(step[1])
    else:
        depth -= int(step[1])
print(h_pos * depth)

# Part two
h_pos = 0
depth = 0
aim = 0
for step in data:
    if step[0] == 'forward':
        h_pos += int(step[1])
        depth += aim * int(step[1])
    elif step[0] == 'down':
        aim += int(step[1])
    else:
        aim -= int(step[1])
print(h_pos * depth)